sci-libs/symengine/symengine-0.1.0.ebuild

```diff
@@ -0,0 +1,53 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+
+PYTHON_COMPAT=( python2_7 )
+PYTHON_REQ_USE=""
+
+inherit cmake-utils python-single-r1
+
+DESCRIPTION="Fast symbolic manipulation library, written in C++"
+HOMEPAGE="https://github.com/sympy/symengine"
+SRC_URI="https://github.com/sympy/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="python threads openmp boost"
+
+DEPEND="
+	dev-libs/jemalloc
+	dev-libs/boost
+	python? ( dev-python/setuptools[${PYTHON_USEDEP}] )"
+RDEPEND="${DEPEND}
+	python? ( dev-python/numpy[${PYTHON_USEDEP}] )"
+
+CMAKE_BUILD_TYPE=Release
+
+src_configure() {
+	local mycmakeargs=(
+		-DCMAKE_INSTALL_PREFIX:PATH="${EPREFIX}"/usr \
+		$(cmake-utils_use_with boost) \
+		$(cmake-utils_use_with python)
+	)
+
+	if use openmp; then
+		if [[ $(tc-getCC) == *gcc ]] && ! tc-has-openmp; then
```

Comment:
@mschubert 
I meant [[ ${MERGE_TYPE != binary ]], see https://devmanual.gentoo.org/ebuild-writing/eapi/index.html. There are occasions like for binary only distros derived from gentoo which do not install any compiler but still are using ebuilds to install packages. On their system that check will always fail in a false positive way if not wrapped by MERGE_TYPE.
Now the question is, if the openmp support is automagic, then I would recommend to hard require it. See the gentoo docs about automagic dependencies and why they are bad.

Secondly, the check should be moved, but into pkg_pretend(). That check is even run before anything starts, see https://devmanual.gentoo.org/ebuild-writing/functions/.